manifests/db.pp

```diff
@@ -28,9 +28,13 @@
   }
   ensure_resource('mysql_database', $dbname, $db_resource)
 
+  case empty($password) {
+    true  : { $password_hash = '' }
+    false : { $password_hash = mysql_password($password) }
```

Comment:
i think puppet lint (or the future parser?) might complain here about a missing default case.